notebooks/official/ray_on_vertex_ai/train/model_training_formalization_with_ray_and_kfp.ipynb

```diff
@@ -0,0 +1,1765 @@
+        "                                        --training_dir={training_dir} \\\n",
+        "                                        --training_run_id={training_run_id} \\\n",
+        "                                        --project_id={project} \\\n",
+        "                                        --region={region} \\\n",
+        "                                        --bucket_uri={bucket_uri}\"\"\",\n",
+        "            cluster_name=cluster_name,\n",
+        "            runtime_env=runtime_env,\n",
+        "        )\n",
+        "        .set_display_name(\"Run Ray Training\")\n",
+        "        .after(get_dataset_uri_task)\n",
+        "    )\n",
+        "\n",
+        "    # get eval metrics task\n",
+        "    get_eval_metrics_task = (\n",
+        "        GetMetricsOp(training_dir=training_dir, training_run_id=training_run_id)\n",
+        "        .set_display_name(\"Get Metrics\")\n",
+        "        .after(run_ray_training_task)\n",
+        "    )\n",
+        "\n",
+        "    # evaluate condition\n",
+        "    with Condition(\n",
+        "        get_eval_metrics_task.outputs[\"threshold_metric\"] < threshold,\n",
+        "        name=\"Blessing condition\",\n",
+        "    ):\n",
+        "\n",
+        "        _ = (\n",
+        "            UploadRayModelOp(\n",
+        "                training_dir=training_dir,\n",
+        "                training_run_id=training_run_id,\n",
+        "                dataset_uri=get_dataset_uri_task.outputs[\"dataset_uri\"],\n",
+        "                model_uri=model_uri,\n",
+        "                explain=explain,\n",
+        "                project=project,\n",
+        "                region=region,\n",
+        "                bucket_uri=bucket_uri,\n",
+        "            )\n",
+        "            .set_display_name(\"Upload Model\")\n",
+        "            .after(get_eval_metrics_task)\n",
+        "        )"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "MLrxk2EY2DD6"
+      },
+      "source": [
+        "### Compile your pipeline into a YAML file\n",
+        "\n",
+        "Compile the pipeline into YAML format. The YAML file includes all the information for executing your pipeline on Vertex AI Pipelines.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "kJKAjQdK2MwE"
+      },
+      "outputs": [],
+      "source": [
+        "compiler.Compiler().compile(pipeline_func=pipeline, package_path=PIPELINE_FILE_PATH)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "dD3nRWMI3nzM"
+      },
+      "source": [
+        "### Run your pipeline\n",
+        "\n",
+        "You use the Vertex AI Python client to submit and run your pipeline as Vertex AI Pipeline job."
```

Comment:
add "a": "You use the Vertex AI Python client to submit and run your pipeline as a Vertex AI Pipeline job."